lec15-git-projects.Rmd

```diff
@@ -0,0 +1,415 @@
+idea to start off your pull request message with an overall summary 
+('adding data cleaning `dplyr` code') followed by a point-form list of
+what changes have been made, if necessary. 
+
+Once the pull request has been made, GitHub will list both your message
+and your commit messages below. You also have the option of merging the
+pull request yourself -- but _don't do this_! When collaborating, always
+have someone else review and merge your pull request. 
+
+If all does not look good, your team members can add messages below, and
+tag others using the @ symbol, similarly to most social networks. If
+more changes are needed before the pull request is ready to merge, 
+any new commits you make to the branch on your fork (`initial-edit` in
+our case here) will automatically be added on to the pull request.
+This way, you can incorporate any changes or fixes suggested by your
+team members simply by continuing to work in your fork's new branch
+until your changes are ready to merge.
+
+Once a pull request has been merged, the branch on our fork isn't needed
+anymore. Because of this, GitHub will immediately prompt us to delete
+the `initial-edit` branch as soon as it's been merged. Our edits
+are now on the main repo!
+
+#### Syncing your fork
+
+There's one issue with the above, though -- now, the `master` branch
+on our fork is out of date with the current state of the main repo.
+To update our fork, we have to go back to the command line.
+
+To begin with, _clone_ your fork to your computer. To do this,
+open a bash shell, and navigate over to whichever folder you would
+like a copy of your repo to be saved in. Then, head to your fork
+on GitHub, and click the green 'Clone or download' button on the right
+side of the page. This yields a link to your repo. Copy this to your clipboard,
+and head back to bash. Then, substituting in your link below, run:
+
+```
+git clone [repo link]
+```
+
+This will create a new folder in your current working directory that contains
+the contents of your fork. Enter this new folder with `cd` and type `git status`
+to make sure the repo has been cloned properly. `git status` should output
+that the branch is even with `origin/master`, indicating that it is currently
+the same as the current state of your fork. 
+
+To get our fork up to date with the main repo, we have to add a _remote_ 
+linking to the main repo. Head to your group's repo and once again click
+on 'Clone or download' to grab its link. Then, using the main repo link, run:
+
+```
+git remote add upstream [repo link]
+```
+
+Once this is done, run:
+
+```
+git remote -v
+```
+
+to get a list of existing remotes. This should return four links,
+two of which are labelled `origin` and two of which are labelled `upstream`.
+
+Next, we have to _fetch_ the new changes that are in the main repo. 
+
+```
+git fetch upstream
+```
+
+Once the edits have been downloaded, merge them into your local repo:
+
+```
+git merge upstream/master
+```
+
```

Comment:
Both commands above can be combined by using `git pull upstream master`. Up to you if you want to mention this.